[BBS-299] Fix leftovers from dropping Python 3.6

pafonta opened the pull request:
## Description

https://github.com/BlueBrain/Search/pull/242 forgot to change things related to Python 3.6:

Fixes [BBS-299](https://bbpteam.epfl.ch/project/issues/browse/BBS-299).

### 1 - The minimal required Python version

https://github.com/BlueBrain/Search/blob/fb1e7885961f70c8f4d9174d7200b758ffda1b67/setup.py#L88

This causes the installation of BBS to fails on Python 3.6 without a clear idea of the reason.

`NumPy` 1.20.1 requires Python >= 3.7. No artefacts are published on PyPI for Python 3.6. When installing BBS on Python 3.6, the following strange error is thrown:

```
ERROR: Could not find a version that satisfies the requirement numpy==1.20.1
ERROR: No matching distribution found for numpy==1.20.1
```

### 2 - The installed Python in `base.Dockerfile`

Installation of `python3.6-dev`.
Mention of `Python 3.6`.
Mention of Docker image `python:3.6`.

### 3 - README

Mention of `Python 3.6`.
Mention of Tox environment `py36`.

## How to test?

### A - Mentions

The command

```
egrep -ri --exclude-dir "./.*" '3\.6|py36' .
```

should ONLY return the following

```
./tests/data/CORD19_samples/biorxiv_medrxiv/biorxiv_medrxiv/pdf_json/9ae476404f7ef1ec1ede965f0b898f31a5bf5a81.json:            "text": "Statistical tests were two-sided with significance set at \u03b1 less than 0.05. We performed all data analyses by R software version 3.6.2 (R Foundation for Statistical Computing).",
./data_and_models/pipelines/sentence_embedding/scripts/build.sh:# Tested on Python 3.6.12 with:
./docs/source/whatsnew.rst:- |Remove| support for :code:`Python 3.6`.
```

### B - base.Dockerfile

The following, adapted from the Getting Started, should finish without errors:

```
export USER_NAME=$(id -un)
export USER_ID=$(id -u)

export http_proxy=http://bbpproxy.epfl.ch:80/
export https_proxy=http://bbpproxy.epfl.ch:80/

git clone --branch bbs_299 https://github.com/BlueBrain/Search.git

export WORKING_DIRECTORY="$(pwd)"
export REPOSITORY_DIRECTORY="$WORKING_DIRECTORY/Search"

cd $REPOSITORY_DIRECTORY
docker build \
  --build-arg BBS_HTTP_PROXY=$http_proxy \
  --build-arg BBS_http_proxy=$http_proxy \
  --build-arg BBS_HTTPS_PROXY=$https_proxy \
  --build-arg BBS_https_proxy=$https_proxy \
  --build-arg BBS_USERS="$USER_NAME/$USER_ID" \
  -f docker/base.Dockerfile -t test_bbs_base_${USER_NAME} .

docker run \
  --volume /raid:/raid \
  --env REPOSITORY_DIRECTORY \
  --gpus all \
  --interactive \
  --tty \
  --rm \
  --user "$USER_NAME" \
  --name test_bbs_base_${USER_NAME} test_bbs_base_${USER_NAME}

cd $REPOSITORY_DIRECTORY

pip install .
```

## Checklist:

- [x] This PR refers to an issue present in the [issue tracker](https://github.com/BlueBrain/Search/issues).
- [x] `setup.py` and `requirements.txt` updated with new dependencies.
- [x] Travis CI pipeline run.

FrancescoCasalegno reviewed (commented):
Nice catch, thank you very much!

I think the following two things also need to be fixed now that we are dropping 3.6 and supporting 3.9:
- [x] `README.md` in a few points
- [ ] `base.Dockerfile` e.g. [here](https://github.com/BlueBrain/Search/blob/bbs_299/docker/base.Dockerfile#L70)

pafonta commented:
Hello @FrancescoCasalegno !

I have addressed the other leftovers you have found.

pafonta commented:
Hello @EmilieDel and @Stannislav !

Thank you for your first review. Since I have addressed the points rose by @FrancescoCasalegno, the PR needs a new review.

Indeed, the installation of Python in `base.Dockerfile` needed to be migrated from 3.6 to 3.7. There was a trick around how to install `pip` for Python 3.7. I have left explanation in the `Dockerfile`.

pafonta commented:
NB: A detailed list of changes and instructions on how to test can be found in the PR's description.

FrancescoCasalegno reviewed (approved):
LGTM, thanks for fixing this nasty issues with the `Dockerfile`!

I would just ask to add in 2 points info about Python 3.9 which is now supported by Blue Brain Search.

jankrepl commented on README.md:
```diff
@@ -608,7 +608,6 @@ rm -R $REPOSITORY_DIRECTORY
- - Python 3.6
  - Python 3.7
  - Python 3.8
 
```
BTW once we are on PyPI we can generate a badge instead
https://shields.io/category/platform-support

jankrepl reviewed (approved):
Thank you for this PR!
